### PR TITLE
feat: `verifyAndParseSignInCallbackUri()`

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

implement `verifyAndParseSignInCallbackUri()`

## Testing

UT

## Reviwers

@logto-io/eng

### CODE REVIEW

File: Sources/Logto/Core/LogtoCore.swift [hunk 1]
```diff
@@ -54,4 +54,32 @@ enum LogtoCore {
...
+            throw LogtoErrors.UriVerification.stateMismatched
+        }
+
+        let errorItems = components.queryItems?.filter { ["error", "error_description"].contains($0.name) } ?? []
+
+        guard errorItems.count == 0 else {
+            throw LogtoErrors.UriVerification.errorItemFound(items: errorItems)
+        }
+
+        guard let code = components.queryItems?.first(where: { $0.name == "code" })?.value else {
```
Comment (simeng-li) [inline Sources/Logto/Core/LogtoCore.swift:79]: interesting point. What if more than one code occurred in the URL 😆 . In js, we use `query-string` to parse the query params. The return type of code was hardcoded as `string`, which in fact, if more than one code are present, will return an array.  cc: @wangsijie @IceHe should we worry about that?

Comment (gao-sun) [reply Sources/Logto/Core/LogtoCore.swift:79]: what's the case that provider will return more than one code?

Comment (IceHe) [reply Sources/Logto/Core/LogtoCore.swift:79]: if return an array, it seems to be a really bad sign —— server has something wrong or is insecure.

A. throw exception
B. select first or last element in the array

I prefer to A

Comment (gao-sun) [reply Sources/Logto/Core/LogtoCore.swift:79]: discussed offline, keep as it is

Comment (IceHe) [reply Sources/Logto/Core/LogtoCore.swift:79]: No need to deal with uri containing more than one `code`.

If more than one, fail by default and keep implementation simple.
We can fix it when needed someday.

Or we may need to concern all query params, which should be a string and maybe an array

**Review by simeng-li — APPROVED** [review]
LGTM

File: Sources/Logto/Core/LogtoCore.swift (hunk 1 above)
Comment (gao-sun) [reply Sources/Logto/Core/LogtoCore.swift:79]: although we do not fail here - the principle is to keep it simple to when parsing query items in SDKs. whether use `.first` or recognize as a single string

**Pull request merged**